### PR TITLE
Add esy support and workarounds

### PR DESCRIPTION
**Pull request by EduardoRFS**
This PR adds support to esy, so that this project can be built by esy during development and also be used by applications directly as shown at https://github.com/anmonteiro/gh-feed-reader/pull/34

It adds a variable BSLIB so that bsc can find the location of the stdlib which on esy is under `bs-platform/lib/bucklescript`

### CODE REVIEW

**Comment by EduardoRFS** [comment]
@anmonteiro can we merge this?

File: fix-install.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#! /bin/sh
```
Comment (anmonteiro) [inline fix-install.sh:1]:
```suggestion
#!/usr/bin/env sh
```

File: fix-install.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+mv others/* .
+mv runtime/* .
+mv stdlib-412/stdlib_modules/* .
+mv stdlib-412/* .
+
+cd $cur__lib
+ln -s $cur__target_dir/default/lib/js .
+ln -s $cur__target_dir/default/lib/es6 .
+
+ln -s $cur__bin/bsc.exe $cur__bin/bsc
```
Comment (anmonteiro) [inline fix-install.sh:17]: feel free to address this in a subsequent PR, but I think we could just fix this in the dune files

```lisp
(install
  (files (bsc.exe as bsc))
  (section bin))
```

**Review by anmonteiro — APPROVED** [review]

File: fix-install.sh (hunk 2 above)
Comment (EduardoRFS) [reply fix-install.sh]: thank you, this works

**Pull request merged**